MonoGame.Framework.Content.Pipeline/MonoGame.Content.Builder.targets

```diff
@@ -11,41 +11,35 @@
     <AvailableItemName Include="MonoGameContentReference" />
   </ItemGroup>
 
-  <!-- This disables the IDE feature that skips executing msbuild in some build situations. --> 
+  <!-- This disables the IDE feature that skips executing msbuild in some build situations. -->
   <PropertyGroup>
     <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
   </PropertyGroup>
 
   <Target Name="Prepare">
-
     <PropertyGroup>
-
-      <ContentRootDirectory Condition=" '$(ContentRootDirectory)' == '' " >Content</ContentRootDirectory>
-      <ParentOutputDir Condition=" '$(ParentOutputDir)' == '' " >$(ProjectDir)$(ContentRootDirectory)\bin\$(MonoGamePlatform)</ParentOutputDir>
-      <ParentIntermediateDir Condition=" '$(ParentIntermediateDir)' == '' " >$(ProjectDir)$(ContentRootDirectory)\obj\$(MonoGamePlatform)</ParentIntermediateDir>
-
       <!-- El Capitan Support -->
       <MonoExe Condition=" '$(OS)' != 'Windows_NT' And Exists ('/Library/Frameworks/Mono.framework/Versions/Current/bin/mono') ">/Library/Frameworks/Mono.framework/Versions/Current/bin/mono</MonoExe>
       <MonoExe Condition=" '$(OS)' != 'Windows_NT' And Exists ('/usr/local/bin/mono') ">/usr/local/bin/mono</MonoExe>
       <MonoExe Condition=" '$(OS)' != 'Windows_NT' And Exists ('/usr/bin/mono') ">/usr/bin/mono</MonoExe>
       <MonoExe Condition=" '$(OS)' != 'Windows_NT' And '$(MonoExe)' == '' ">mono</MonoExe>
-
       <MonoGameContentBuilderExe Condition="'$(MonoGameContentBuilderExe)' == ''">$(MSBuildThisFileDirectory)Tools\MGCB.exe</MonoGameContentBuilderExe>
       <MonoGameContentBuilderCmd>&quot;$(MonoGameContentBuilderExe)&quot;</MonoGameContentBuilderCmd>
       <MonoGameContentBuilderCmd Condition=" '$(OS)' != 'Windows_NT' ">$(MonoExe) $(MonoGameContentBuilderCmd)</MonoGameContentBuilderCmd>
-
       <PlatformResourcePrefix Condition="'$(MonoGamePlatform)' == 'MacOSX' Or '$(MonoGamePlatform)' == 'iOS'">Resources\</PlatformResourcePrefix>
       <PlatformResourcePrefix Condition="'$(MonoGamePlatform)' == 'Android'">Assets\</PlatformResourcePrefix>
       <PlatformResourcePrefix Condition="'$(PlatformResourcePrefix)' == ''"></PlatformResourcePrefix>
-
-      <Header>/platform:$(MonoGamePlatform) /outputDir:&quot;$(ParentOutputDir)&quot; /intermediateDir:&quot;$(ParentIntermediateDir)&quot; /quiet</Header>
+      <Header>/platform:$(MonoGamePlatform) </Header>
     </PropertyGroup>
 
     <!-- Get all Mono Game Content References and store them in a list -->
     <!-- We do this here so we are compatible with xbuild -->
     <ItemGroup>
-      <ContentReferences Include="@(MonoGameContentReference)"/>
-     </ItemGroup>
+      <ContentReferences Include="@(MonoGameContentReference)">
+        <ContentOutputDir>$(ProjectDir)%(RelativeDir)\bin\$(MonoGamePlatform)\%(Filename)</ContentOutputDir>
+        <ContentIntermediateOutputDir>$(ProjectDir)%(RelativeDir)\obj\$(MonoGamePlatform)\%(Filename)</ContentIntermediateOutputDir>
```

Comment:
It seems RelativeDir and Filename are sometimes not correctly when using xbuild.
xbuild isn't very helpful in its error message:
`Error building target Prepare: Object reference not set to an instance of an object`

... That's it :/ I only figured out that RelativeDir and Filename are the culprits by putting some messages around it.

Another issue is that MonoDevelop is very bad at handling links. Or rather it doesn't create links at all. It just includes the file like it normally would... So you have to dig into the msbuild files to get that right.